### PR TITLE
Add support for .NET Framework 4.6.1

### PR DESCRIPTION
**Pull request by FabienLavocat**
Add the support for .NET Framework 4.6.1 from GitHub Issue #4976

### CODE REVIEW

**Comment by rseanhall** [comment]
New packages need to be added to the documentation at src/chm/documents/customactions/wixnetfxextension.html.md

File: src/ext/NetFxExtension/wixlib/NetFx461.wxs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        Official documentation can be found at the following location:
+
+           .NET Framework 4.5/4.5.1/4.5.2/4.6/4.6.1 - http://msdn.microsoft.com/en-us/library/w0x726c2(v=vs.110).aspx
+    -->
+
+  <?define NetFx461MinRelease = 394254 ?>
+  <?define NetFx461WebLink = http://go.microsoft.com/fwlink/?LinkId=671728 ?>
+  <?define NetFx461RedistLink = http://go.microsoft.com/fwlink/?LinkId=671743 ?>
+  <?define NetFx461EulaLink = https://msdn.microsoft.com/en-us/library/ee942965.aspx ?>
```
Comment (rseanhall) [inline src/ext/NetFxExtension/wixlib/NetFx461.wxs]: The EULA link needs to be a direct link to the EULA, not a .NET information page.

**Comment by FabienLavocat** [comment]
I've updated the documentation and the link to the EULA.

**Comment by jaydubbyasee** [comment]
Is there anything currently preventing this pull request from being merged? I need this for one of my projects.

**Comment by robmen** [comment]
Still waiting for @FabienLavocat to complete the Assignment Agreement at: http://wixtoolset.org/development/assignment-agreement/

**Comment by FabienLavocat** [comment]
I did it the minute you gave me the link. I can do it one more time

**Comment by robmen** [comment]
Yeah, we've had some issues with that process lately. :sad: Hopefully, kicking it again will make it work the second time around.

**Comment by FabienLavocat** [comment]
No problem, I did it again 2 minutes ago.

**Comment by Seve0212a** [comment]
What is the ETA that this will complete and available?

**Comment by FabienLavocat** [comment]
@robmen did you receive my agreement?

**Comment by Mawe9999** [comment]
Any news on this?

**Comment by FabienLavocat** [comment]
@Mawe9999 I have signed the agreement, it shouldn't be long until someone merge it into the main code.

**Comment by xaicon** [comment]
Tried to use a RegistrySearch to check the Release value for my .NET 4.6.1 condition. Now I understand the need for this variable. The Dword returns '#' with the Release value. This does not work for comparison like it does when I use Util:registrySearch for bundles. I am very interested in getting this included. Sounds like the work is done. Please add :) Thanks.

**Comment by cadahl** [comment]
This is probably blocking a fair number of people, including us. Please merge soon. :)

**Comment by curtnichols** [comment]
We could use this ASAP.  :)

**Comment by FabienLavocat** [comment]
As a temporary workaround, what you can do is to add the file https://github.com/FabienLavocat/wix3/blob/develop/src/ext/NetFxExtension/wixlib/NetFx461.wxs to your Wix Bundle and reference it in the Bundle > Chain with: `<PackageGroupRef Id="NetFx461Redist" />`

**Comment by Mawe9999** [comment]
@FabienLavocat works, thank you.

**Comment by jaydubbyasee** [comment]
What are the legal implications for including this file directly in our project? Would I have to release the source code for my installer/bootstrapper according to the Microsoft Reciprocal License?

**Comment by FabienLavocat** [comment]
I don't think so because you do not modify the source code of the wix project. But I might be wrong.
I hope the Wix team can merge it quickly into the main project

**Comment by jaydubbyasee** [comment]
Thanks really do hope this gets merged soon as well.

**Comment by robmen** [comment]
One more week at the most... working out our assignment agreement issues now.

**Comment by carlsonad** [comment]
I have included the NetFx461.wxs in my wix burn project and it works well except on a clean Windows 2012r2 machine.  In that case the .net framework fails to install.  The problem ended up being that the 4.6.1 framework requires KB2919355.  

Finding that this is the problem is not straight forward.   The netfx install reports a generic installation error with a link to the boot strapper log file.  In that log file it will point you to the netfx log file which is a html file.  In that log file you'll see the following error:

> The update corresponding to KB2919355 needs to be installed before you can install this product on Windows 8.1 or Windows Server 2012 R2.

Is there a way to make this more user friendly?

**Comment by FabienLavocat** [comment]
Yes, the KB2919355 has to be installed first on Windows Server 2012 R2.
Is it possible in WIX to make sure that KB is installed before running an installer?

**Comment by stavnskov** [comment]
I don't think it is that simple. It is 6-7 downloads that needs to be installed in correct order. A total of 1 GB download. I have also seen a Win 8.1 PC with NET4.6.1 but without the update!

**Comment by robmen** [comment]
@FabienLavocat now that we've moved to .NET Foundation, can you [sign the .NET Foundation CLA](http://wixtoolset.org/development/assignment-agreement/). I promise this is the last time you'll need to go through a process... and the .NET Foundation one is fully automated so it should go quickly.

Sorry for all the license trouble. This PR hit us in transition that is finally complete. Thanks.

**Comment by FabienLavocat** [comment]
It's signed :)

**Comment by barnson** [comment]
Merged with 64c5335. Thanks for all the paperwork! :)

**Comment by FabienLavocat** [comment]
Thanks!

**Pull request closed without merging**